### PR TITLE
CI: Upload site assets

### PR DESCRIPTION
**Pull request by Wxh16144**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [ ] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [x] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

ref: https://github.com/ant-design/ant-design/pull/49213/files#r1625446496

### 💡 Background and Solution

之前一直想搞一个 docker 镜像，然后提交了一个 PR #49213 ， 但是感觉还是麻烦了一些， antd 仓库改用 GitHub Release 页面上传静态站点。

原因是，我在反馈 #51430 问题时候，明显可以启动一个指定版本的 website 进行对比，但是我自己每次都要拉指定版本源码，安装依赖然后进行构建，其次为了方便其他开发者可以直接订阅 release hooks 然后下载 release asset 进行其他部署之类的。

比起 CD 的 artifact （默认 90天） , Release 可以永久保留，不过组织存储资源用量需要 @afc163  确认一下。仅此～

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      --     |
| 🇨🇳 Chinese |      --     |

### CODE REVIEW

**Comment by afc163** [comment]
直接访问 https://github.com/ant-design/ant-design/commit/f456147e4da24d5e6c8a3862c1cfd9af327fef48#commitcomment-148158051 这里的 surge 链接如何？

**Review by afc163 — APPROVED** [review]

File: .github/workflows/site-deploy.yml [hunk 1]
```diff
@@ -66,3 +83,26 @@ jobs:
...
+
+      - name: Tarball site
+        run: |
+          cd ./_site
+          VERSION=$(echo ${{ github.ref_name }} | sed 's/\./-/g')
+          tar -czf ../website.tar.gz --transform 's|^|antd-${VERSION}-website/|' .
+          cd ..
+
+      - name: Upload to Release
+        uses: softprops/action-gh-release@c062e08bd532815e2082a85e87e3ef29c3e6d191 # v2.0.8
```
Comment (afc163) [inline .github/workflows/site-deploy.yml:105]: 更新成 2.1.0？

Comment (Wxh16144) [reply .github/workflows/site-deploy.yml:105]: 2.0.8 能用，不过还得你添加到白名单，或者得开发到 action-cool 组织下面？

Comment (afc163) [reply .github/workflows/site-deploy.yml:105]: 已经加了

**Pull request merged**